subprojects/gradle/build-trace/src/gradleTest/kotlin/com/avito/android/build_trace/CriticalPathTest.kt

```diff
@@ -0,0 +1,280 @@
+package com.avito.android.build_trace
+
+import com.avito.android.build_trace.internal.critical_path.CriticalPathSerialization
+import com.avito.test.gradle.TestResult
+import com.avito.test.gradle.gradlew
+import com.google.common.truth.Truth.assertThat
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+
+class CriticalPathTest {
+
```

Comment:
Integration tests. 

They are needed mostly to check the correct conversion executed tasks for the algorithm.